src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -721,24 +722,23 @@ export class PostgresStorageAdapter {
     });
     const qs = `CREATE TABLE IF NOT EXISTS $1:name (${patternsArray.join(',')})`;
     const values = [className, ...valuesArray];
-    return conn.task(t => {
-      return this._ensureSchemaCollectionExists(t)
-        .then(() => t.none(qs, values))
-        .catch(error => {
-          if (error.code === PostgresDuplicateRelationError) {
-            // Table already exists, must have been created by a different request. Ignore error.
-          } else {
+    
+    return conn.task('create-table', function * (t) {
+      try {
+        yield self._ensureSchemaCollectionExists(t);
+        yield t.none(qs, values);
+      } catch(error) {
+          if (error.code !== PostgresDuplicateRelationError) {
             throw error;
-          }})
-    })
-      .then(() => {
-        return conn.tx('create-relation-tables', t => {
-          const queries = relations.map((fieldName) => {
-            return t.none('CREATE TABLE IF NOT EXISTS $<joinTable:name> ("relatedId" varChar(120), "owningId" varChar(120), PRIMARY KEY("relatedId", "owningId") )', {joinTable: `_Join:${fieldName}:${className}`});
-          });
-          return t.batch(queries);
-        });
+          }
+        // ELSE: Table already exists, must have been created by a different request. Ignore the error.
+      }
+      yield t.tx('create-table-tx', tx => {
+        return tx.batch(relations.map(fieldName => {
```

Comment:
@vitaly-t Merry Christmas. Quick question. I've notice you got rid of `t.batch` in favor of `t.none(this._pgp.helpers(...))`. Is there a reason why you use it here.

Reply:
@dplewis as per that PR, it is a performance optimization, to execute all queries via a single IO operation, as opposed to executing them separately. It is much faster that way ;)

Reply:
I always thought `.batch` was a single I/O.

Reply:
Looks good to me 👍

Reply:
`batch` is an operation irrelevant as to the IO, it simply resolves an array of promises, like `Promise.all` does, except it also settles all the promises.